ouroboros-consensus/src/Ouroboros/Storage/ChainDB/API.hs

```diff
@@ -224,8 +224,10 @@ toChain chainDB = bracket
     go chain it = do
       next <- iteratorNext it
       case next of
-        IteratorExhausted  -> return chain
-        IteratorResult blk -> go (Chain.addBlock blk chain) it
+        IteratorResult blk  -> go (Chain.addBlock blk chain) it
+        IteratorExhausted   -> return chain
+        IteratorBlockGCed _ ->
+          error "block on the current chain was garbage-collected"
```

Comment:
In theory this can happen due to timing, even though in practice this is extrordinarily unlikely because this function is eager so will not trigger the GC corner case that requires one to use it very very slowly.

But since it's only used for tests, it's fine. Still, since people may copy'n'paste, no harm in using `throwM $ BlockGCed @blk hash` or whatever.

Reply:
You have the `BlockGCed` error but don't seem to use it here?